app/src/main/java/com/example/watpool/services/FirebaseService.kt

```diff
@@ -59,8 +72,110 @@ class FirebaseService : Service() {
         return coordinateService.fetchCoordinatesByLocation(latitude, longitude, radiusInKm)
     }
 
-    fun addCoordinate(driverId: String, latitude: Double, longitude: Double): Task<DocumentReference> {
-        return coordinateService.addCoordinate(driverId, latitude, longitude)
+    fun addCoordinate(driverId: String, latitude: Double, longitude: Double, location: String = ""): Task<DocumentReference> {
+        return coordinateService.addCoordinate(driverId, latitude, longitude, location)
+    }
+
+    fun fetchUsersByUsername(username: String) : Task<QuerySnapshot> {
+        return userService.fetchUsersByUsername(username)
+    }
+    fun createUser(username: String, name: String, phone: String) : Task<DocumentReference> {
+        return userService.createUser(username, name, phone)
+    }
+    fun createDriver(username: String, licenseNumber: String, carModel: String, carColor: String ) : Task<Void> {
+        return userService.createDriver(username, licenseNumber, carModel, carColor)
+    }
+
+//    fun cloneTrip(tripId: String) {
+//        tripsService.cloneTrip(tripId)
+//    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun createTrip(driverId: String, startingCoordinateId: String, endingCoordinateId: String, tripDate: LocalDate, maxPassengers: String, isRecurring: Boolean = false, recurringDayOfTheWeek: FirebaseTripsService.DayOfTheWeek = FirebaseTripsService.DayOfTheWeek.SUNDAY, recurringEndDate: LocalDate = LocalDate.now()): Task<DocumentReference> {
+        return tripsService.createTrip(driverId, startingCoordinateId, endingCoordinateId, tripDate, maxPassengers, isRecurring, recurringDayOfTheWeek, recurringEndDate)
+    }
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun createTripPosting(userId: String, startingCoordinateId: String, endingCoordinateId: String, tripDate: LocalDate, isRecurring: Boolean = false, recurringDayOfTheWeek: FirebaseTripsService.DayOfTheWeek = FirebaseTripsService.DayOfTheWeek.SUNDAY, recurringEndDate: LocalDate = LocalDate.now()): Task<DocumentReference> {
+        return tripsService.createTripPosting(userId, startingCoordinateId, endingCoordinateId, tripDate, isRecurring, recurringDayOfTheWeek, recurringEndDate)
+    }
+    fun createTripConfirmation(tripId: String, confirmationDate: LocalDate, riderIds: List<String>): Task<DocumentReference> {
+        return tripsService.createTripConfirmation(tripId, confirmationDate, riderIds)
+    }
+
+    private fun tripsCoordinateConnector(task: Task<QuerySnapshot>): Task<List<DocumentSnapshot>> {
```

Comment:
@FredLiu876 This function will auto fetch the coordinates and add the location names to the trips